test cases/fortran/7 generated/meson.build

```diff
@@ -7,12 +7,22 @@ project('generated', 'fortran',
 conf_data = configuration_data()
 conf_data.set('ONE', 1)
 conf_data.set('TWO', 2)
-conf_data.set('THREE', 3)
 
-configure_file(input : 'mod3.fpp', output : 'mod3.f90', configuration : conf_data)
-# Manually build absolute path to source file to test
-# https://github.com/mesonbuild/meson/issues/7265
-three = library('mod3', meson.current_build_dir() / 'mod3.f90')
+mod3_f = custom_target(
```

Comment:
Changing this from a configure_file to a custom_target is fine, I'm sure, but... I don't think custom_target does absolute paths, so are we not dropping the coverage for the issue referenced there?

Why are you also changing the library() invocation to use mod3_f instead of `meson.current_build_dir() / 'mod3.f90'`?

Reply:
`meson.corrent_build_dir()` was a workaround for the mentioned bug, not the bug itself. So the test wasn't actually doing anything useful in this case. I had changed it to do the correct/idiomatic thing in the first commit since that bug was fixed but the test was never updated. I guess we could build two copies of mod3.f90 or something?

Reply:
Well, as long as the original issue got fixed in the meantime I guess. :)